### PR TITLE
feat(space): add subDID to root-store entry

### PR DESCRIPTION
**Pull request by oed**
Puts the subDIDs in the root-store entry that also has the space odbAddress. Also adding a `type` to the root-store entry.

### CODE REVIEW

File: src/space.js [hunk 1]
```diff
@@ -122,15 +132,6 @@ class Space {
...
   }
-
-  async _ensureDIDPublished () {
-    // Ensure we self-published our did
-    if (!(await this.public.get('proof_did'))) {
-      await this._syncSpacePromise
-      // we can just sign an empty JWT as a proof that we own this DID
-      await this.public.set('proof_did', await this._3id.signJWT({}, { space: this._name }), { noLink: true })
-    }
-  }
```
Comment (zachferland) [inline src/space.js]: 👍

File: src/space.js [hunk 2]
```diff
@@ -37,8 +44,12 @@ class Space {
       const spaceAddress = await this._store._load()
 
       const entries = await this._rootStore.iterator({ limit: -1 }).collect()
-      if (!entries.find(entry => entry.payload.value.odbAddress.indexOf(nameToSpaceName(this._name)) !== -1)) {
-        this._rootStore.add({ odbAddress: spaceAddress })
+      const entry = entries.find(entry => entry.payload.value.odbAddress.indexOf(nameToSpaceName(this._name)) !== -1)
+      if (!entry) {
+        await this._rootStore.add({ type: 'space', DID: this.DID, odbAddress: spaceAddress })
```
Comment (zachferland) [inline src/space.js]: 👍

Comment (zachferland) [reply src/space.js]: then in future, rootstore <-> space/did link can be verified by resolving the did now? right?

Comment (oed) [reply src/space.js]: Precisely :)

**Pull request merged**